method_comparison/peft_bench/utils.py

```diff
@@ -0,0 +1,453 @@
+    if benchmark_result.run_info.get("status") != BenchmarkStatus.SUCCESS.value:
+        print_fn(f"Error: {benchmark_result.run_info.get('error', 'Unknown error')}")
+        # Optionally print other sections if needed for failed runs, or just return
+        print_fn("=" * 50)
+        return
+
+    print_fn("\nModel Information:")
+    print_fn(f"  Base Model: {benchmark_result.meta_info.get('model_id', 'N/A')}")
+
+    print_fn("\nParameter Counts:")
+    params = benchmark_result.meta_info.get("parameters", {})
+    print_fn(f"  Base Parameters: {params.get('base_params', 0):,}")
+    print_fn(f"  Trainable Parameters: {params.get('trainable_params', 0):,}")
+    print_fn(f"  Parameter Ratio: {params.get('param_ratio', 0):.5%}")
+
+    print_fn("\nModel Size:")
+    size_info = benchmark_result.meta_info.get("model_size", {})
+    print_fn(f"  Base Model: {size_info.get('base_model_size_mb', 0):.2f} MB")
+    print_fn(f"  Adapter: {size_info.get('adapter_size_mb', 0):.2f} MB")
+
+    print_fn("\nMemory Usage (from generation_info):")
+    memory_data = benchmark_result.generation_info.get("memory", {})
+    print_fn(f"  Peak GPU Memory: {memory_data.get('peak_gpu_memory_mb', 0):.2f} MB")
+    print_fn(f"  Peak RAM Memory: {memory_data.get('peak_ram_memory_mb', 0):.2f} MB")
+
+    print_fn("\nDetailed Metrics (from generation_info.by_category):")
+    if benchmark_result.generation_info.get("by_category"):
+        for category, cat_data in benchmark_result.generation_info["by_category"].items():
+            print_fn(f"  Category: {category}")
+            metrics = cat_data.get("metrics", {})
+            print_fn(f"    Inference Time: {metrics.get('inference_time', 0):.4f} seconds")
+            print_fn(f"    Base Inference Time: {metrics.get('base_inference_time', 0):.4f} seconds")
+            print_fn(f"    Inference Overhead: {metrics.get('inference_overhead_pct', 0):.2f}%")
+            print_fn(f"    Time Per Token: {metrics.get('time_per_token', 0):.6f} seconds/token")
+            print_fn(f"    Generated Tokens: {metrics.get('generated_tokens', 0):.1f}")
+
+            # Print sample statistics if available
+            samples = cat_data.get("samples", [])
+            if samples:
+                print_fn(f"    Number of Samples: {len(samples)}")
+                print_fn(f"    Average Generated Tokens: {sum(s.get('generated_tokens', 0) for s in samples) / len(samples):.1f}")
+    else:
+        print_fn("  No per-category metrics available.")
+
+    benchmark_result.compute_overall_metrics()
+
+    print_fn("\nOverall Metrics (from generation_info.overall):")
+    if benchmark_result.generation_info.get("overall"):
+        for metric_name, value in benchmark_result.generation_info["overall"].items():
+            if isinstance(value, float):
+                print_fn(f"  {metric_name.replace('_', ' ').title()}: {value:.4f}")
+            else:
+                print_fn(f"  {metric_name.replace('_', ' ').title()}: {value}")
```

Comment:
I like that the "Detailed Metrics" are nicely formatted, e.g.

```
    Inference Overhead: 3.32%
    Time Per Token: 0.014478 seconds/token
    ...
```

but the "Overall Metrics" don't have nice formatting:

```
  Inference Time: 0.2893
  Base Inference Time: 0.2827
  Inference Overhead Pct: 2.3610
  Time Per Token: 0.0145
  Generated Tokens: 20.0000
```

Let's format them in the same nice way.